module-apps/apps-common/widgets/ProgressTimerWithBarGraphAndCounter.cpp

```diff
@@ -18,11 +18,31 @@ namespace app
     {
         updateText();
         updateProgress();
+        updateTimeWidget();
         ProgressTimer::update();
     }
 
     void ProgressTimerWithBarGraphAndCounter::updateText()
     {
+        using utils::time::Duration;
+        if (text == nullptr) {
+            return;
+        }
+        const auto secondsRemaining = duration - std::chrono::duration_cast<std::chrono::seconds>(elapsed);
+        const Duration remainingDuration{std::time_t{secondsRemaining.count()}};
```

Comment:
consider adding constructor in the `Duration` class for `std::chrono`'s object

Reply:
The code was restored form inital commit introducing(deleting this code) TimeFixedWidget so I did not want to change anything. The benefit from new constructor would be lack of casting to time_t however I would like to not introduce changes in Duration as its working and tested